lib/load-paths-handler.coffee

```diff
@@ -4,33 +4,45 @@ path = require 'path'
 _ = require 'underscore-plus'
 {GitRepository} = require 'atom'
 {Minimatch} = require 'minimatch'
+{GitProcess} = require 'dugite'
 
 PathsChunkSize = 100
 
 emittedPaths = new Set
 
 class PathLoader
-  constructor: (@rootPath, ignoreVcsIgnores, @traverseSymlinkDirectories, @ignoredNames) ->
+  constructor: (@rootPath, @ignoreVcsIgnores, @traverseSymlinkDirectories, @ignoredNames) ->
     @paths = []
     @realPathCache = {}
-    @repo = null
-    if ignoreVcsIgnores
-      repo = GitRepository.open(@rootPath, refreshOnWindowFocus: false)
-      @repo = repo if repo?.relativize(path.join(@rootPath, 'test')) is 'test'
 
   load: (done) ->
-    @loadPath @rootPath, true, =>
-      @flushPaths()
-      @repo?.destroy()
-      done()
+    repo = GitRepository.open(@rootPath, refreshOnWindowFocus: false)
+    if repo?.relativize(path.join(@rootPath, 'test')) is 'test'
+      args = ['ls-files', '-c', '-o', '-z']
```

Comment:
Would you mind using the extended command args for clarity?  Eg `--cached`, `--others`, etc.